code/modules/mob/living/carbon/xenomorph/xenomorph.dm

```diff
@@ -63,6 +63,17 @@
 	ADD_TRAIT(src, TRAIT_BATONIMMUNE, TRAIT_XENO)
 	ADD_TRAIT(src, TRAIT_FLASHBANGIMMUNE, TRAIT_XENO)
 
+	//For triggering hive alerts when we take damage.
+	RegisterSignal(src, COMSIG_XENOMORPH_TAKING_DAMAGE, .proc/damage_taken)
+
+/mob/living/carbon/xenomorph/proc/damage_taken()
+	SIGNAL_HANDLER
```

Comment:
Why isnt this inlined

Reply:
What do you mean by this exactly?

Reply:
I why isn't this in the actual xeno take damage proc